model/task/task.go

```diff
@@ -1745,6 +1740,108 @@ func (t *Task) Archive() error {
 	return nil
 }
 
+func ArchiveMany(tasks []Task) error {
+	if len(tasks) == 0 {
+		return nil
+	}
+	// add execution tasks of display tasks passed in, if they are not there
+	execTaskMap := map[string]*Task{}
+	for _, t := range tasks {
+		execTaskMap[t.Id] = &t
+	}
+	additionalTasks := []string{}
+	for _, t := range tasks {
+		for _, et := range t.ExecutionTasks {
+			if execTaskMap[et] == nil {
+				additionalTasks = append(additionalTasks, t.ExecutionTasks...)
+				continue
+			}
+		}
+	}
+	if len(additionalTasks) > 0 {
+		toAdd, err := Find(ByIds((additionalTasks)))
```

Comment:
Find iterates over all tasks to make sure there are no display tasks, but we already know there won't be based on what we put in additionalTasks, so you should be able to use FindAll and skip that iteration